pkg/querier/distributor_queryable.go

```diff
@@ -108,9 +108,9 @@ func (q *distributorQuerier) Select(sortSeries bool, sp *storage.SelectHints, ma
 		)
 
 		if q.streamingMetadata {
-			ms, err = q.distributor.MetricsForLabelMatchersStream(ctx, model.Time(q.mint), model.Time(q.maxt), matchers...)
+			ms, err = q.distributor.MetricsForLabelMatchersStream(ctx, model.Time(minT), model.Time(maxT), matchers...)
```

Comment:
I am not sure about this change. We updated the `minT` and `maxT` on L94 to times in select hints, but we are still using the querier time. Is it a bug or we intended to do so?

Reply:
Humm..

Can we check if the sp.Start and sp.End are populate for `/series` and `/labels` calls?

Reply:
Ok.. seems fine.
https://github.com/prometheus/prometheus/blob/6b53aeb012080ab2a50acd229bfe9943125abfa6/web/api/v1/api.go#L770-L775